programs/conditional_vault/src/lib.rs

```diff
@@ -155,6 +155,104 @@ pub mod conditional_vault {
         Ok(())
     }
 
+    pub fn merge_conditional_tokens_for_underlying_tokens(
+        ctx: Context<MergeConditionalTokensForUnderlyingTokens>, amount: u64
+    ) -> Result<()> {
+        let accs = &ctx.accounts;
+
+        let vault = &accs.vault;
+        let vault_status = vault.status;
+        require!(vault_status == VaultStatus::Active, ErrorCode::VaultAlreadySettled);
```

Comment:
you already have this covered in the constraints

Reply:
Yeh, I know it's double checked but I think the check should be happening here instead. I suppose I can just delete it.